toptal/picasso
# fix: remove tags from headings

[FX-4222]

### Description

This pull request fixes two bugs. 

The first bug can be observed when user selects part of the heading and uses hotkey to set it as bold or italic. The resulting HTML should not contain those tags.

The second bug can be observed when converting a regular string (some parts of it should be bold or italic) into heading. The newly converted heading should not convert bold or italic tags.

In general, headings **should never contain** any formatting tags.

### How to test


- [Temploy](https://picasso.toptal.net/fx-4222-rte-using-ctrl-b-outputs-strong-element-in-the-html-output)
- Try converting text with **bold** and _italic_ fragments to heading – **bold** and _italic_ format should be discarded
- Try **bold** and _italic_ hot keys when selecting heading – there should be no effect on heading

### Screenshots

Before

https://github.com/toptal/picasso/assets/1390758/a7aaef4f-d71f-4faf-b8fc-7fb17860c8e9

After

https://github.com/toptal/picasso/assets/1390758/68bde1e4-be47-4bdb-a274-101bbb562e07


### Development checks

- [N/A] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [N/A] Annotate all `props` in component with documentation
- [N/A] Create `examples` for component


> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[FX-4222]: https://toptal-core.atlassian.net/browse/FX-4222?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
